### ISSUE TITLE
Helm: /data/stan/log: permission denied for cluster

### ISSUE THREAD

**Issue by kylos101** [nats-io/k8s#52]
Hi,

I tried to install stan like this:

```
helm repo add nats https://nats-io.github.io/k8s/helm/charts/
helm install stan nats/stan \
--set stan.nats.url=nats://nats-client.default.svc.cluster.local:4222 \
--set store.cluster.enabled=true
```

It starts, and provisions a persistent volume, but the logs indicate a permission denied error:

```
[1] 2020/05/03 17:50:46.354899 [INF] STREAM: Starting nats-streaming-server[stan] version 0.17.0
[1] 2020/05/03 17:50:46.354992 [INF] STREAM: ServerID: e4CikyxM5FvXzfJ1EiIQdU
[1] 2020/05/03 17:50:46.354995 [INF] STREAM: Go version: go1.13.7
[1] 2020/05/03 17:50:46.354998 [INF] STREAM: Git commit: [f4b7190]
[1] 2020/05/03 17:50:46.398915 [INF] STREAM: Recovering the state...
[1] 2020/05/03 17:50:46.401310 [INF] STREAM: Recovered 1 channel(s)
[1] 2020/05/03 17:50:46.401401 [INF] STREAM: Cluster Node ID : stan-0
[1] 2020/05/03 17:50:46.401445 [INF] STREAM: Cluster Log Path: /data/stan/log
[1] 2020/05/03 17:50:46.401570 [INF] STREAM: Shutting down.
[1] 2020/05/03 17:50:46.401961 [FTL] STREAM: Failed to start: mkdir /data/stan/log: permission denied
```

As a workaround, I am opting to store logs as a sub-folder in the same spot as store.file.path (which resolves to /data/stan/store) like this:

```
helm install stan nats/stan \
--set store.type=file --set store.file.storageSize=1Gi \
--set stan.nats.url=nats://nats-client.default.svc.cluster.local:4222 \
--set store.cluster.enabled=true --set store.cluster.logPath=/data/stan/store/log
```

I'm not sure if there's a better solution, but wanted to share here to discuss. Thanks!

**Comment by wallyqs** [comment]
This is due to this issue: https://github.com/nats-io/k8s/issues/48

You can set the following instead as a workaround, but will make release soon.

```yaml
securityContext: null
```

**Comment by kylos101** [comment]
Hi @wallyqs I was able to work around the problem by overriding store.cluster.logPath like below. Refer to the last line (--set) for the override.

```
helm install stan nats/stan \
--set store.type=file \
--set store.file.storageSize=1Gi \
--set stan.nats.url=nats://nats-client.default.svc.cluster.local:4222 \
--set store.cluster.enabled=true \
--set store.cluster.logPath=/data/stan/store/log
```

In other words, I didn't set securityContext, got it working using the above example, but am unclear why /data/stan/log yields a permission error. I am guessing it is because /data/stan/log is a path that does't exist on the pod..but that's just a guess.

Thanks for the heads up about an upcoming release. I'll circle back to this and test again once it's released.

**Comment by etix** [comment]
AFAICT this is solved in v0.4.0

**Comment by kylos101** [comment]
> AFAICT this is solved in v0.4.0

That is correct, I'll close this issue.
